### PR TITLE
airbyte-ci: factor git checkout

### PR DESCRIPTION
**Pull request by postamar**
refactor in anticipation of future changes

### CODE REVIEW

**Comment by postamar** [comment]
Current dependencies on/for this PR:
* master
  * **PR #32801** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/32801?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #32802** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/32802?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #32803** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/32803?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #32804** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/32804?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/32801?utm_source=stack-comment).

File: airbyte-ci/connectors/pipelines/pipelines/helpers/git.py [hunk 1]
```diff
@@ -118,6 +80,37 @@ def get_modified_files_in_pull_request(pull_request: PullRequest) -> List[str]:
     return [f.filename for f in pull_request.get_files()]
 
 
+async def checked_out_git_container(
```
Comment (bnchrch) [inline airbyte-ci/connectors/pipelines/pipelines/helpers/git.py:83]: 💅 Could we move this to `airbyte-ci/connectors/pipelines/pipelines/dagger/containers`

e.g. `airbyte-ci/connectors/pipelines/pipelines/dagger/containers/git.py`

**Review by bnchrch — APPROVED** [review]
One small nit but nothing to hold this back.

LGTM!

**Review by alafanechere — APPROVED** [review]
👍 
FYI I deeply consolidated the `local` vs `remote` checkout  and diff operations in the context of my hackday project. This project is an attempt to make airbyte-ci able to target any repo. I wont merge this work anytime soon (until this features hits our roadmap).

https://github.com/airbytehq/airbyte/pull/32703

**Pull request closed without merging**